contrib/windows-cmake/CMakeLists.txt

```diff
@@ -64,11 +121,30 @@ add_library(libhwloc
     ${TOPDIR}/hwloc/topology-xml-nolibxml.c
     ${TOPDIR}/hwloc/topology-windows.c
     ${TOPDIR}/hwloc/topology-x86.c
+    $<$<BOOL:${HWLOC_HAVE_LIBXML2}>:${TOPDIR}/hwloc/topology-xml-libxml.c>
+    $<$<BOOL:${HWLOC_HAVE_OPENCL}>:${TOPDIR}/hwloc/topology-opencl.c>
+    $<$<BOOL:${HAVE_CUDA}>:${TOPDIR}/hwloc/topology-cuda.c>
+)
+
+target_link_libraries(hwloc PRIVATE
+    $<$<BOOL:${HWLOC_HAVE_LIBXML2}>:LibXml2::LibXml2>
+    $<$<BOOL:${HWLOC_HAVE_OPENCL}>:OpenCL::OpenCL>
+    "$<$<BOOL:${HAVE_CUDA}>:CUDA::cudart;CUDA::cuda_driver>"
+)
+
+if(BUILD_SHARED_LIBS)
```

Comment:
Modern CMake would use a Cache variable `HWLOC_BUILD_SHARED_LIBS` and set its default to `BUILD_SHARED_LIBS` -> see for example vtk/paraview